### PR TITLE
Update flake8 to the latest version and fix all errors

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

**Comment by hugovk** [comment]
Merged into the fork here: https://github.com/html5lib-python/html5lib-python/pull/1

**Comment by gsnedders** [comment]
FWIW, I have a preference in favour of pinning flake8 and its dependencies to stop tests from starting failing without changes in html5lib, though preferably in combination with adding some bot to update pinned dependencies.

**Comment by jdufresne** [comment]
No problem. I've changed the PR to keep flake8 pinned, but now to the latest version. All errors are still fixed.

**Comment by hugovk** [comment]
> preferably in combination with adding some bot to update pinned dependencies.

The two bots I know about: dependabot, now part of GitHub; and pyupbot, which I hear has been a bit flaky of late.

**Comment by gsnedders** [comment]
Yeah, we use pyupbot on web-platform-tests and it seems to have been going downhill…

File: html5lib/_utils.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
         # We need this with u"" because of http://bugs.jython.org/issue2039
         _x = eval('u"\\uD800"')  # pylint:disable=eval-used
         assert isinstance(_x, text_type)
-except:  # pylint:disable=bare-except
+except Exception:  # pylint:disable=bare-except
```
Comment (gsnedders) [inline html5lib/_utils.py]: Comment is now wrong (though maybe if we  #292 not worth keeping around? but I guess we will?).

Comment (jdufresne) [reply html5lib/_utils.py]: Thanks for catching. Fixed in the latest revision.

**Pull request merged**